crates/rspack_plugin_javascript/src/visitors/dependency/common_js_export_scanner.rs

```diff
@@ -209,16 +270,41 @@ impl<'a> CommonJsExportDependencyScanner<'a> {
     }
   }
 
+  fn is_module_exports_member_expr_start(&self, mut expr: &Expr) -> bool {
+    loop {
+      match expr {
+        _ if expr_matcher::is_module_exports(expr) => return true,
+        Expr::Member(MemberExpr { obj, .. }) => expr = obj.as_ref(),
+        _ => return false,
+      }
+    }
+  }
+
+  fn is_this_member_expr_start(&self, mut expr: &Expr) -> bool {
+    if self.enter_call != 0 {
+      return false;
+    }
+    loop {
+      match expr {
+        _ if self.is_this_expr(expr) => return true,
+        Expr::Member(MemberExpr { obj, .. }) => expr = obj.as_ref(),
+        _ => return false,
+      }
+    }
+  }
+
   fn is_exports_or_module_exports_or_this_expr(&self, expr: &Expr) -> bool {
-    matches!(expr,  Expr::Ident(ident) if &ident.sym == "exports" && ident.span.ctxt == self.unresolved_ctxt)
-      || expr_matcher::is_module_exports(expr)
-      || matches!(expr,  Expr::This(_) if  self.enter_call == 0)
+    self.is_exports_expr(expr) || expr_matcher::is_module_exports(expr) || self.is_this_expr(expr)
   }
 
   fn is_exports_expr(&self, expr: &Expr) -> bool {
     matches!(expr,  Expr::Ident(ident) if &ident.sym == "exports" && ident.span.ctxt == self.unresolved_ctxt)
   }
 
+  fn is_this_expr(&self, expr: &Expr) -> bool {
+    matches!(expr,  Expr::This(_) if self.enter_call == 0 && self.stmt_level == 1)
```

Comment:
Use stmt level to check top level this

Reply:
top level check is hard, for example

```javascript
if (condition) {
  this.foo = 1;
}
```

Reply:
You are right, perhaps I may check top level just like [SideEffectsFlagPlugin](https://github.com/web-infra-dev/rspack/blob/main/crates/rspack_plugin_javascript/src/plugin/side_effects_flag_plugin.rs#L115-L147)?